### ISSUE TITLE
Translation and i18n plans

### ISSUE THREAD

**Issue by dabura667** [keybase/client#7368]
Are there any? May I help? Are you opposed as a project to translating at this point in the project?

**Comment by dabura667** [comment]
Also, Which repos should I look at to translate each client (I'm assuming this one repo will be sufficient?) and the website (I'm guessing it's in a different repo?)

**Comment by dabura667** [comment]
I am currently maintaining the Crowdin (translation site for hosting and volunteering to translate open source apps for free) pages for 4 major bitcoin applications, and have translated all of them into Japanese.

I am pushing for my friends here in Japan to start using the app, but English is a big hurdle for them.

Definitely would like to get some feedback as to thoughts on i18n before I dedicate time to doing anything just to get a big "oops, we don't wanna do that right now, sorry."

**Comment by luixxiul** [comment]
+1 on localization into Japanese.

**Comment by heini** [comment]
Yes, this is definitely something that is missing, and, from my p.o.v., preventing a more widespread adoption of KB. There are still man people in the world who don't speak english.

**Comment by kovacsdavid** [comment]
+1 we need this in hungarian too :)

**Comment by FraYoshi** [comment]
+2 for japanese and italian :)

**Comment by MineMy** [comment]
+1 for korean and deutsch.
I hope more people use keybase :D

**Comment by np** [comment]
+1 for french

**Comment by mathieujobin** [comment]
since the acquisition, we can only hope for the best.

I'd say, anyone with experience with translating electron app could give it a shot at setting up the base to be able to translate the app.

open a PR. and people can add their translations on top.

hopefully, it will get merged

**Comment by mathieujobin** [comment]
As far as I can see there are no i18n tags around text element https://github.com/keybase/client/blob/cac9573e33f472fcb1417c1e6a899bfbba36405c/shared/settings/account/index.tsx#L70

So unless it can be implemented within the `<Kb.text ..>` tag, someone will need to wrap all text in a function such as `t()` or `_()` depending on the standard being adopted.

**Comment by itsnotsaved** [comment]
2017 👉 ⏰ 😵 🤒 ☹️ 2021
I think if someone able to collect source strings and organize codebase for i18n better and i don't think that's hard task for keybase maintainers too. For translations, https://hosted.weblate.org/ is a opensource project and cloud edition free for libre projects. it's possible to set-up, configure it easily and set auto sync translation to this repository then forget it.

**Comment by mathieujobin** [comment]
I thought we could try https://github.com/lukasgeiter/gettext-extractor to extract all english strings into a pot file. then use poedit service to automate the translation.

I am getting an error with gettext-extractor, if anyone wants to help and look at my issue

otherwise, we need to manually replace all strings with something like this

https://ttag.js.org/docs/typescript.html
